docs/example_docstrings.py

```diff
@@ -0,0 +1,460 @@
+in the end look like in the compiled format.
+
+Once the your code is properly documented, it will be included using
+`autodoc <https://www.sphinx-doc.org/en/master/usage/extensions/autodoc.html>`_ directives
+into the global structure of the PorePy documentation.
+This will be done in a next step. Before that, familiarize yourself with the various directives
+and their options beforehand.
+
+"""
+from __future__ import annotations
+
+from typing import Generator, Literal, Optional
+
+import numpy as np
+import scipy.sparse as sps
+
+import porepy as pp
+
+
+module_level_var_1: str = "var"
+"""Variables at the module level are documented inline after declaration."""
+
+
+module_level_var_2: int = 1
+"""Multi-line docstrings are also valid.
+
+You might want to use multiline docstrings when a more detailed description
+of the variable is required.
+
+In every case, the closing \"\"\" of a multiline docstring must be on a separate line after a
+blank line.
+
+**DO:**
+
+.. code:: Python
+
+    var: int = 1
+    \"\"\"Introduction line.
+
+    Line 2.
+
+    \"\"\"
+
+**DO NOT:**
+
+.. code:: Python
+
+    var: int = 1
+    \"\"\"Introduction line.
+
+    Line 2.\"\"\"
+
+.. code:: Python
+
+    var: int = 1
+    \"\"\"Introduction line.
+
+    Line 2.
+    \"\"\"
+
+"""
+
+
+module_level_var_3: int = 42
+"""int: **DO NOT** start the docstring with ``int:``, i.e. with type hints.
+
+This will render the below ``Type: int``, which is unnecessary due to the already present
+annotation after the variable name. As you can test above, clicking on *int* will forward you
+to the right documentation.
+
+**DO:**
+
+.. code:: Python
+
+    module_level_var_3: int = 1
+    \"\"\"This is var 3.\"\"\"
+
+**DO NOT:**
+
+.. code:: Python
+
+    module_level_var_3 = 1
+    \"\"\"int: This is var 3.\"\"\"
+
+"""
+
+
+def module_level_function_1(arg1: int, arg2: str, arg3: bool) -> bool:
+    """A short description of the functionality must fit in the first line.
+
+    Detailed description can be given next if necessary. The extended description
+    can of course span several lines.
+
+    When natural, the description can be break into several paragraphs that start
+    without indentation.
+
+    The description of arguments is started using the Google Style directive *Parameters:*.
+
+    .. note::
+
+        You can find the ``[source]`` link in the upper right corner of this function's
+        documentation and take a look at the Python and rst source code of this docstring,
+        as well as for every docstring that follows.
+
+    The return value can be described using the Google Style directive *Returns:*.
+
+    Parameters:
+        arg1: Description of arg1.
+        arg2: Description of arg2. If the description of a parameter spans more than one line,
+            the lines must be indented.
+
+            You can even use multiple paragraphs for describing a parameter. There
```

Comment:
This can be modified by introducing custom HTML/CSS settings. We need to compromise between readability in the docstrings, which Eirik emphasizes very often, and how we format the resulting HTML/CSS